File: deepgoal/core/pipeline/pipe.py
from __future__ import annotations

from ..container import PipelineItem, SerialNode
from ..node import NodeResult
from ..types import EngineOptions


class Pipe(SerialNode):
    """顺序执行一组 Node/Executor，Pipe 本身也是 Node，可嵌套进任何容器。"""

    def __init__(self, items: list[PipelineItem], options: EngineOptions | None = None) -> None:
        super().__init__(items)
        self._options = options or EngineOptions()

    async def run(self, input_file_path: str, options: EngineOptions | None = None) -> NodeResult:
        return await super().run(input_file_path, options or self._options)

File: deepgoal-ts/src/core/pipeline/pipe.ts
import { SerialNode, type PipelineItem } from '../container.js';
import type { PipelineInput, PipelineResult } from '../node.js';
import { createEngineOptions, type EngineOptions } from '../types/index.js';

export class Pipe extends SerialNode {
  private readonly defaultOptions: EngineOptions;

  constructor(items: PipelineItem[], options: EngineOptions = createEngineOptions()) {
    super(items);
    this.defaultOptions = options;
  }

  override async run(input: PipelineInput, options: EngineOptions = this.defaultOptions): Promise<PipelineResult> {
    return super.run(input, options);
  }
}
